core/src/main/java/com/radixdlt/api/system/routes/DbCheckpointHandler.java

```diff
@@ -0,0 +1,119 @@
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.api.system.routes;
+
+import com.google.inject.Inject;
+import com.radixdlt.api.system.SystemJsonHandler;
+import com.radixdlt.api.system.generated.models.CreateDbCheckpointResponse;
+import com.radixdlt.db.checkpoint.RustDbCheckpoints;
+import com.radixdlt.store.NodeStorageLocation;
+import com.radixdlt.utils.properties.RuntimeProperties;
+import java.io.File;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public final class DbCheckpointHandler extends SystemJsonHandler<CreateDbCheckpointResponse> {
+
+  private final boolean enabled;
+  private final String checkpointsPath;
+  private final RustDbCheckpoints rustCheckpoints;
+
+  @Inject
+  DbCheckpointHandler(
+      RustDbCheckpoints rustCheckpoints,
+      RuntimeProperties runtimeProperties,
+      @NodeStorageLocation String nodeStorageLocation) {
+    this.enabled = runtimeProperties.get("api.system.enable_db_checkpoint", false);
+
+    this.checkpointsPath =
+        runtimeProperties.get(
+            "db.checkpoints_path",
+            Path.of(nodeStorageLocation, "state_manager_checkpoints").toString());
```

Comment:
_(aesthetics-only)_ looks like our usual approach is "handle the system props in the Guice modules + pass `*Config` objects to the logic".

Reply:
good point, done